assets/js/i18n/LocaleSwitcher.js

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+import { withI18n } from '@lingui/react'
+import { supportedLocales, setLocale } from './locales'
+
+const LocaleSwitcher = ({ i18n }) => {
+  if (window.i18n && window.i18n.showSwitcher) {
```

Comment:
`i18n` is an object provided by LinguiJs that handles translations.
`window.i18n` are the settings we pass from django (current locale, whether to show switcher, etc); I thought it should be better to group them together.
Looks like we should rename `window.i18n` to something like `window.i18n_settings` for clarity. What do you think?

Reply:
Thanks for explaining.

I agree we should leave the LinguiJS name, `i18n`. As for `window.i18n`: I don't like `window.i18nSettings` or `window.i18nConfig` because today, those global variables contain _code_. (Somewhere down the line, that may change; but not today.) Other options are synonyms -- `window.localeData`, etc.; but synonyms add more confusion than they remove.

Personally, I think it's okay to call both, `i18n`. Please add a doc-string to the `LocaleSwitcher` component explaining what's in the `window.i18n` variable; and please add PropTypes to the `LocaleSwitcher` component with a comment on the `i18n` prop explaining that it is a LinguiJS `i18n` object.

I'm not terribly opinionated about what we name things; but I think the comments are important.

Reply:
D'oh I got this wrong! `window.i18n` doesn't include messages. So yeah, let's call it `window.i18nConfig`.